public/app/plugins/datasource/cloudwatch/language_provider.test.ts

```diff
@@ -0,0 +1,171 @@
+  });
+
+  it('should suggest fields and some functions after `by` command', async () => {
+    await runSuggestionTest('stats count(something) by \\', [
+      fields,
+      STRING_FUNCTIONS.concat(DATETIME_FUNCTIONS, IP_FUNCTIONS).map(v => v.label),
+    ]);
+  });
+
+  it('should suggest fields and some functions after comparison operator', async () => {
+    await runSuggestionTest('filter field1 >= \\', [
+      fields,
+      BOOLEAN_FUNCTIONS.map(v => v.label),
+      NUMERIC_OPERATORS.map(v => v.label),
+    ]);
+  });
+
+  it('should suggest fields directly after sort', async () => {
+    await runSuggestionTest('sort \\', [fields]);
+  });
+
+  it('should suggest fields directly after sort after a pipe', async () => {
+    await runSuggestionTest('fields field1 | sort \\', [fields]);
+  });
+
+  it('should suggest sort order after sort command and field', async () => {
+    await runSuggestionTest('sort field1 \\', [['asc', 'desc']]);
+  });
+
+  it('should suggest fields directly after parse', async () => {
+    await runSuggestionTest('parse \\', [fields]);
+  });
+
+  it('should suggest fields and bool functions after filter', async () => {
+    await runSuggestionTest('filter \\', [fields, BOOLEAN_FUNCTIONS.map(v => v.label)]);
+  });
+});
+
+async function runSuggestionTest(query: string, expectedItems: string[][]) {
+  const result = await getProvideCompletionItems(query);
+  expectedItems.forEach((items, index) => {
+    expect(result.suggestions[index].items.map(item => item.label)).toEqual(items);
+  });
+}
+
+function makeDatasource(): CloudWatchDatasource {
+  return {
+    getLogGroupFields(): Promise<GetLogGroupFieldsResponse> {
+      return Promise.resolve({ logGroupFields: [{ name: 'field1' }, { name: '@message' }] });
+    },
+  } as any;
+}
+
+/**
+ * Get suggestion items based on query. Use `\\` to mark position of the cursor.
+ */
+function getProvideCompletionItems(query: string): Promise<TypeaheadOutput> {
+  const provider = new CloudWatchLanguageProvider(makeDatasource());
+  const cursorOffset = query.indexOf('\\');
+  const queryWithoutCursor = query.replace('\\', '');
+  let tokens: Token[] = Prism.tokenize(queryWithoutCursor, provider.getSyntax()) as any;
+  tokens = addTokenMetadata(tokens);
+  const value = new ValueMock(tokens, cursorOffset);
+  return provider.provideCompletionItems(
+    {
+      value,
+    } as any,
+    { logGroupNames: ['logGroup1'] }
+  );
+}
+
+class ValueMock {
+  selection: Value['selection'];
+  data: Value['data'];
+
+  constructor(tokens: Array<string | Token>, cursorOffset: number) {
+    this.selection = {
+      start: {
+        offset: cursorOffset,
+      },
+    } as any;
+
+    this.data = {
+      get() {
+        return tokens;
+      },
+    } as any;
+  }
+}
+
+/**
+ * Adds some Slate specific metadata
+ * @param tokens
+ */
+function addTokenMetadata(tokens: Array<string | Token>): Token[] {
```

Comment:
Not sure if this is the best way to do this. Seems it may be a bit error prone long term, trying to emulate slate but seems to work for now and maybe could be used in general for autocomplete testing.